src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py

```diff
@@ -37,3 +47,18 @@ def validate_reservation_summary(namespace):
         raise CLIError("usage error: --grain  can be either daily or monthly.")
     if data_grain == 'daily' and (not namespace.start_date or not namespace.end_date):
         raise CLIError("usage error: Both --start-date and --end-date need to be supplied for daily grain.")
+
+
+def validate_budget_parameters(namespace):
+    """lowercase the time grain for comparison"""
+    budgetcategory = namespace.category.lower()
+    if budgetcategory != 'cost' and budgetcategory != 'usage':
+        raise CLIError("usage error: --parameters.category must be set to either cost or usage")
```

Comment:
The param name is `--category`. User won't know what `--parameters.category` is.

Reply:
Additionally, this should be covered by using `get_enum_type` in the param registration.
https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#registering-enums